programs/mango-v4/src/instructions/flash_loan.rs

```diff
@@ -377,6 +377,10 @@ pub fn flash_loan_end<'key, 'accounts, 'remaining, 'info>(
     let mut token_loan_details = Vec::with_capacity(changes.len());
     for (change, oracle_price) in changes.iter().zip(oracle_prices.iter()) {
         let mut bank = health_ais[change.bank_index].load_mut::<Bank>()?;
+        if change.amount.abs() > 0 {
+            require!(!bank.is_reduce_only(), MangoError::SomeError);
+        }
```

Comment:
I don't think this check is right. It says: any change at all is forbidden for reduce_only banks. But using flash loan to repay borrows or withdraw deposits should be allowed!

Reply:
done

Reply:
Suggestion for reduce-only:
- if change < 0 && native + change < 0: error
- if change > 0 && native + change >= 1: error